File: app/src/main/java/com/crevion/apps/cleanandroidcode/deps/Deps.java
package com.crevion.apps.cleanandroidcode.deps;

import com.crevion.apps.cleanandroidcode.home.HomeActivity;
import com.crevion.apps.cleanandroidcode.networking.NetworkModule;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by yusufaw on 1/27/18.
 */

@Singleton
@Component(modules = {NetworkModule.class,})
public interface Deps {
    void inject(HomeActivity homeActivity);
}

File: app/src/main/java/com/crevion/apps/cleanandroidcode/deps/Deps.kt
package com.crevion.apps.cleanandroidcode.deps

import com.crevion.apps.cleanandroidcode.ui.home.HomeActivity
import com.crevion.apps.cleanandroidcode.networking.NetworkModule
import com.crevion.apps.cleanandroidcode.preferences.PreferencesModule
import dagger.Component
import javax.inject.Singleton

/**
 * Created by yusufaw on 2/9/18.
 */

@Singleton
@Component(modules = [PreferencesModule::class, NetworkModule::class])
interface Deps {
    fun inject(homeActivity: HomeActivity)
}